clients/database/request_metric.go

```diff
@@ -34,8 +34,13 @@ type ProxiedRequestMetric struct {
 }
 
 // Save saves the current ProxiedRequestMetric to
-// the database, returning error (if any)
+// the database, returning error (if any).
+// If db is nil, returns nil error.
 func (prm *ProxiedRequestMetric) Save(ctx context.Context, db *bun.DB) error {
+	if db == nil {
+		return nil
+	}
```

Comment:
nitpick: any reason we don't want to return an explicit error rather than a silent return?

Reply:
Yeh, this is to avoid refactoring. The implementation we have in not methods-based, but expects that we pass the db. To remove db dependency during initialization we return not fake, but nil db and pass it. As it is expected (and added to the comments), we allow to pass db as nil. In such case we don't need errors. 

Another ways to do that: 
- switch to interface and add `empty` implementation, that does nothing
- return particulr errors, but in this case we need to refactor processing among the codebase

This helps us to keep the current metrics in the code, while we don't have the decision about about how do we want to implement metrics

Reply:
There is an additional further PR (already created), that we discussed with @pirtleshell and in the task: https://app.shortcut.com/kava-labs/story/14336
This one provide a better refactoring, adding interfaces.